The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/4932872056/jobs/8816301819) [(plain)](https://github.com/rust-lang/rust/commit/ae6910d863213aa6062292938a5abf11d2f1c99a/checks/13362566649/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
    Checking object v0.31.1
    Checking miniz_oxide v0.6.2
    Checking std_detect v0.1.5 (/checkout/library/stdarch/crates/std_detect)
    Checking addr2line v0.19.0
error[E0599]: no method named `derived_type` found for reference `&object::pe::ImageSymbol` in the current scope
  --> library/std/src/../../backtrace/src/symbolize/gimli/coff.rs:58:20
   |
58 |             if sym.derived_type() != object::pe::IMAGE_SYM_DTYPE_FUNCTION || section_number == 0 {
   |                    ^^^^^^^^^^^^ method not found in `&ImageSymbol`
   = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
   |
   |
1  + use object::coff::ImageSymbol;


error[E0599]: no method named `name` found for reference `&'a object::pe::ImageSymbol` in the current scope
   --> library/std/src/../../backtrace/src/symbolize/gimli/coff.rs:102:27
    |
102 |         self.symbols[i].1.name(self.strings).ok()
    |                           ^^^^ field, not a method
    = help: items from traits can only be used if the trait is in scope
help: remove the arguments
    |
    |
102 -         self.symbols[i].1.name(self.strings).ok()
102 +         self.symbols[i].1.name.ok()
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
    |
    |
1   + use object::coff::ImageSymbol;

For more information about this error, try `rustc --explain E0599`.
error: could not compile `std` (lib) due to 2 previous errors
Build completed unsuccessfully in 0:00:14
```

</details>

--- a/compiler/rustc_codegen_llvm/Cargo.toml
+++ b/compiler/rustc_codegen_llvm/Cargo.toml
@@ -11,7 +11,7 @@ bitflags = "1.0"
 cstr = "0.2"
 libc = "0.2"
 measureme = "10.0.0"
-object = { version = "0.30.1", default-features = false, features = [
+object = { version = "0.31.1", default-features = false, features = [
     "std",
     "read",
 ] }

--- a/compiler/rustc_codegen_ssa/Cargo.toml
+++ b/compiler/rustc_codegen_ssa/Cargo.toml
@@ -14,7 +14,7 @@ itertools = "0.10.1"
 tracing = "0.1"
 jobserver = "0.1.22"
 tempfile = "3.2"
-thorin-dwp = "0.4"
+thorin-dwp = "0.6"
 pathdiff = "0.2.0"
 serde_json = "1.0.59"
 snap = "1"
@@ -46,7 +46,7 @@ rustc_session = { path = "../rustc_session" }
 libc = "0.2.50"
 
 [dependencies.object]
-version = "0.30.1"
+version = "0.31.1"
 default-features = false
 features = ["read_core", "elf", "macho", "pe", "unaligned", "archive", "write"]
 

--- a/src/bootstrap/Cargo.toml
+++ b/src/bootstrap/Cargo.toml
@@ -37,7 +37,7 @@ filetime = "0.2"
 cc = "1.0.69"
 libc = "0.2"
 hex = "0.4"
-object = { version = "0.29.0", default-features = false, features = ["archive", "coff", "read_core", "unaligned"] }
+object = { version = "0.31.1", default-features = false, features = ["archive", "coff", "read_core", "unaligned"] }
 serde = "1.0.137"
 # Directly use serde_derive rather than through the derive feature of serde to allow building both
 # in parallel and to allow serde_json and toml to start building as soon as serde has been built.

--- a/src/tools/tidy/src/deps.rs
+++ b/src/tools/tidy/src/deps.rs
@@ -107,6 +107,7 @@ const PERMITTED_RUSTC_DEPENDENCIES: &[&str] = &[
     "autocfg",
     "bitflags",
     "block-buffer",
+    "byteorder", // via ruzstd in object in thorin-dwp
     "cc",
     "cfg-if",
     "chalk-derive",
@@ -206,6 +207,7 @@ const PERMITTED_RUSTC_DEPENDENCIES: &[&str] = &[
     "rustc-rayon",
     "rustc-rayon-core",
     "rustc_version",
+    "ruzstd", // via object in thorin-dwp
     "ryu",
     "scoped-tls",
     "scopeguard",

